File: packages/java/anip-service/src/main/java/dev/anip/service/CapabilityDef.java
package dev.anip.service;

import dev.anip.core.CapabilityDeclaration;

import java.util.Map;
import java.util.function.BiFunction;

/**
 * Binds a capability declaration to a handler function.
 */
public class CapabilityDef {

    private final CapabilityDeclaration declaration;
    private final BiFunction<InvocationContext, Map<String, Object>, Map<String, Object>> handler;

    public CapabilityDef(CapabilityDeclaration declaration,
                         BiFunction<InvocationContext, Map<String, Object>, Map<String, Object>> handler) {
        this.declaration = declaration;
        this.handler = handler;
    }

    public CapabilityDeclaration getDeclaration() {
        return declaration;
    }

    public BiFunction<InvocationContext, Map<String, Object>, Map<String, Object>> getHandler() {
        return handler;
    }
}

File: packages/csharp/src/Anip.Service/CapabilityDef.cs
using Anip.Core;

namespace Anip.Service;

/// <summary>
/// Binds a capability declaration to a handler function.
/// </summary>
public class CapabilityDef
{
    public CapabilityDeclaration Declaration { get; }
    public Func<InvocationContext, Dictionary<string, object?>, Dictionary<string, object?>> Handler { get; }

    public CapabilityDef(
        CapabilityDeclaration declaration,
        Func<InvocationContext, Dictionary<string, object?>, Dictionary<string, object?>> handler)
    {
        Declaration = declaration;
        Handler = handler;
    }
}
